File: leet_DP_189_rotate_array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        L = len(nums)
        k %= L
        if k == 0:
            return
        i = L - k
        A = []
        while i < L:
            A.append(nums[i])
            i += 1
        i = 0
        while i < L - k:
            A.append(nums[i])
            i += 1
        nums[:] = A

File: leet_DP_189_rotate_array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k)
    {
        vector<int> res;
        int         n, i;
        
        n = nums.size();
        k %= n;
        if (!k)
            return;
        i = n - k - 1;
        while (++i < n)
            res.push_back(nums[i]);
        i = -1;
        while (++i < n - k)
            res.push_back(nums[i]);
        nums = res;
    }
};
